types/underscore/index.d.ts

```diff
@@ -3814,150 +3814,144 @@ declare module _ {
         propertyOf(object: object): (key: string | number | Array<string | number>) => any;
 
         /**
-        * Performs an optimized deep comparison between the two objects,
-        * to determine if they should be considered equal.
-        * @param object Compare to `other`.
-        * @param other Compare to `object`.
-        * @return True if `object` is equal to `other`.
-        **/
+         * Performs an optimized deep comparison between the two objects
+         * to determine if they should be considered equal.
+         * @param object Compare to `other`.
+         * @param other Compare to `object`.
+         * @returns True if `object` is equal to `other`.
+         **/
         isEqual(object: any, other: any): boolean;
 
         /**
-        * Returns true if object contains no values.
-        * @param object Check if this object has no properties or values.
-        * @return True if `object` is empty.
-        **/
+         * Returns true if an enumerable object contains no values (no enumerable own-properties).
```

Comment:
Basically, you can call the parameter `collection` and say "returns true if the collection has no elements". The next line about strings and arrays can stay as is. We have updated the documentation to make it clearer that `isEmpty` is meant for collections, but this hasn't made its way into a new release yet.

(As an aside, "enumerable" as an adjective only applies to properties, not to objects.)

Reply:
Updated!